lib/action_controller/serialization.rb

```diff
@@ -18,48 +19,37 @@ def serialization_scope
         respond_to?(_serialization_scope, true)
     end
 
-    def get_serializer(resource)
-      @_serializer ||= @_serializer_opts.delete(:serializer)
-      @_serializer ||= ActiveModel::Serializer.serializer_for(resource)
-
-      if @_serializer_opts.key?(:each_serializer)
-        @_serializer_opts[:serializer] = @_serializer_opts.delete(:each_serializer)
+    def get_serializer(resource, options = {})
+      if ! use_adapter?
+        warn "ActionController::Serialization#use_adapter? has been removed. "\
+          "Please pass 'adapter: false' or see ActiveSupport::SerializableResource#serialize"
+        options[:adapter] = false
       end
-
-      @_serializer
-    end
-
-    def use_adapter?
-      !(@_adapter_opts.key?(:adapter) && !@_adapter_opts[:adapter])
-    end
-
-    [:_render_option_json, :_render_with_renderer_json].each do |renderer_method|
-      define_method renderer_method do |resource, options|
-        @_adapter_opts, @_serializer_opts =
-          options.partition { |k, _| ADAPTER_OPTION_KEYS.include? k }.map { |h| Hash[h] }
-
-        if use_adapter? && (serializer = get_serializer(resource))
-          @_serializer_opts[:scope] ||= serialization_scope
-          @_serializer_opts[:scope_name] = _serialization_scope
-
+      ActiveModel::SerializableResource.serialize(resource, options) do |serializable_resource|
+        if serializable_resource.serializer?
+          serializable_resource.serialization_scope ||= serialization_scope
+          serializable_resource.serialization_scope_name = _serialization_scope
           begin
-            serialized = serializer.new(resource, @_serializer_opts)
+            serializable_resource.adapter
           rescue ActiveModel::Serializer::ArraySerializer::NoSerializerError
```

Comment:
This doesn't seem to be the right place or way to catch the failure when instantiating the serializer